.github/scripts/manage-vm.py

```diff
@@ -253,30 +306,37 @@ def create_vm(sdk, args):
             github_output("local-ipv4", local_ipv4)
             if external_ipv4:
                 github_output("external-ipv4", external_ipv4)
+
+            logger.info("Waiting for VM to be registered as Github Runner")
+            runner_id = wait_for_runner_registration(
+                gh, instance_id, args.github_repo_owner, args.github_repo, 10
+            )
+            if runner_id is not None:
```

Comment:
You can just raise an exception while waiting for the runner registration. And log runner_id there also.

Reply:
I do not see runner_id used in other places, but if it is not used except for logging you can return None.

Reply:
I, initally wanted to call it in `remove_vm` also, so I decided to move this decision, to throw exception or not, to different level.